duesseldorf.py

```diff
@@ -0,0 +1,163 @@
+import re
+from urllib.request import urlopen
+
+from bs4 import BeautifulSoup as parse
+
+from pyopenmensa.feed import OpenMensaCanteen
+from utils import Parser
+
+# this will match notes, e.g. "[R]" or "(1,10,12)"
+legend_regex = re.compile('[\[(][ARFSVZ\d\,]+[\])]')
+
+# this list was compiled from pdf available at http://www.stw-d.de/wp-content/uploads/2017/01/Zeichenerklaerung.pdf
+legend_dict = {'R': 'mit Rindfleisch',
+               'S': 'mit Schweinefleisch',
+               'A': 'mit Alkohol',
+               'F': 'fleischlos (enthält tierische Erzeugnisse)',
+               'V': 'vegan',
+               'Z': 'Bitte beachten Sie die Auszeichnung am Produkt',
+               '1': 'Mit Konservierungsmittel',
+               '2': 'mit Antioxidationsmittel',
+               '3': 'mit Farbstoff',
+               '4': 'mit Geschmacksverstärker',
+               '5': 'mit Schwefel',
+               '6': 'mit Phosphat',
+               '7': 'geschwärzt',
+               '8': 'gewachst',
+               '9': 'mit Süßungsmittel',
+               '10': 'enthält eine Phenylalaninquelle',
+               '11': 'mit Säurungsmittel',
+               '12': 'mit Stabilisatoren',
+               '13': 'mit Phosphorsäure',
+               '14': 'mit Nitritpökelsalz',
+               '15': 'mit Milcheiweiß',
+               '16': 'koffeinhaltig',
+               '17': 'chininhaltig',
+               '18': 'enthält Schwefeldioxid und Sulfite',
+               '19': 'Milch und Milcherzeugnisse / enthält Laktose',
+               '20': 'Glutenhaltiges Getreide sowie daraus hergestellte Erzeugnisse',
+               '21': 'Soja und Sojaerzeugnisse',
+               '22': 'Sellerie und Sellerieerzeugnisse',
+               '23': 'Senf und Senferzeugnisse',
+               '24': 'Sesamsamen und Sesamerzeugnisse',
+               '25': 'Lupine und Lupinenerzeugnisse',
+               '26': 'Erdnüsse und Erdnusserzeugnisse',
+               '27': 'Fisch und Fischerzeugnisse',
+               '28': 'Krebstiere und Krebstiererzeugnisse',
+               '29': 'Weichtiere und Weichtiererzeugnisse',
+               '30': 'Schalenfrüchte und Schalenfruchterzeugnisse',
+               '31': 'Eier und Eierzeugnisse'
+               }
+
+# strings to be removed from meal title
+remove_strings = ['- Preis je 100g:',
+                  '- Preis ab:']
+
+
+def parse_url(url, today=False):
+    canteen = OpenMensaCanteen()
+    document = parse(urlopen(url).read())
```

Comment:
Looking at the commits by @mswart from today (0e39dc58a4a1d4cc9267f277eea225f1e3ef50d3), we should probably explicitly specify the parser here.